### PR TITLE
Skip outline if the draw operation fills with the same colour

### PR DESCRIPTION
**Pull request by radarhere**
If the outline colour in a draw operation is the same as the fill colour, skip drawing the outline.

### CODE REVIEW

**Comment by hugovk** [comment]
Do we have regression tests to cover these changes where the outline and fill colours are the same?

File: src/PIL/ImageDraw.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def _getink(self, ink, fill=None):
                 if self.palette and not isinstance(ink, numbers.Number):
                     ink = self.palette.getcolor(ink)
                 ink = self.draw.draw_ink(ink, self.mode)
-            if fill is not None:
+            if fill is not None and ink != fill:
```
Comment (wiredfool) [inline src/PIL/ImageDraw.py]: Fill and ink are variable type values, e.g. 'white' and (255,255,255) are equivalent, but not equal. You can only really compare them for equality after the ImageColor.getcolor call.

Comment (radarhere) [reply src/PIL/ImageDraw.py]: Okay, thanks. I've removed that line change.

**Comment by radarhere** [comment]
Okay, I've added tests.

**Pull request merged**